src/coreclr/vm/exceptionhandlingqcalls.h

```diff
@@ -19,8 +19,8 @@ extern "C" void QCALLTYPE ResumeAtInterceptionLocation(REGDISPLAY* pvRegDisplay)
 extern "C" void QCALLTYPE AppendExceptionStackFrame(QCall::ObjectHandleOnStack exceptionObj, SIZE_T ip, SIZE_T sp, int flags, ExInfo *pExInfo);
 extern "C" BOOL QCALLTYPE EHEnumInitFromStackFrameIterator(StackFrameIterator *pFrameIter, IJitManager::MethodRegionInfo *pMethodRegionInfo, EH_CLAUSE_ENUMERATOR * pEHEnum);
 extern "C" BOOL QCALLTYPE EHEnumNext(EH_CLAUSE_ENUMERATOR* pEHEnum, RhEHClause* pEHClause);
-extern "C" bool QCALLTYPE SfiInit(StackFrameIterator* pThis, CONTEXT* pStackwalkCtx, bool instructionFault, bool* pIsExceptionIntercepted);
-extern "C" bool QCALLTYPE SfiNext(StackFrameIterator* pThis, unsigned int* uExCollideClauseIdx, bool* fUnwoundReversePInvoke, bool* pIsExceptionIntercepted);
+extern "C" BOOL QCALLTYPE SfiInit(StackFrameIterator* pThis, CONTEXT* pStackwalkCtx, BOOL instructionFault, /* .NET bool */ uint8_t* pIsExceptionIntercepted);
+extern "C" BOOL QCALLTYPE SfiNext(StackFrameIterator* pThis, unsigned int* uExCollideClauseIdx, /* .NET bool */ uint8_t* fUnwoundReversePInvoke, /* .NET bool */ uint8_t* pIsExceptionIntercepted);
```

Comment:
```suggestion
extern "C" BOOL QCALLTYPE SfiNext(StackFrameIterator* pThis, unsigned int* uExCollideClauseIdx, CLR_BOOL* fUnwoundReversePInvoke, CLR_BOOL* pIsExceptionIntercepted);
```
We have a type for this.

It does not look great to use a mix of two different bool types in a single signature. I would either use `BOOL` for everything (change the arg type to `out bool` so that the marshalling kicks in) or `CLR_BOOL` for everything.

Reply:
I thought the preference is standard type names. Maybe here it makes sense as it saves comment explanation.

> or `CLR_BOOL` for everything.

One byte variant, U1. 👍 C23 made `bool` the first-class citizen (without requiring `#include <stdbool.h>`) with `BOOL_WIDTH 1 // exact value`.

Reply:
> C23 made bool the first-class citizen

Do they guarantee that the underlying representation is 0/1? Roslyn expects the underlying bool representation to be 0/1: https://github.com/dotnet/roslyn/blob/main/docs/compilers/Boolean%20Representation.md .

Reply:
Apparently so: false=0 and true=1 https://godbolt.org/z/dcPvzvTzW. Under  `E.2 Minimum values` https://www.open-std.org/jtc1/sc22/wg14/www/docs/n3220.pdf, it states <kbd>#define BOOL_WIDTH 1 // exact value</kbd> and <kbd>#define BOOL_MAX 1 // 2<sup>BOOL_WIDTH</sup> − 1</kbd>.

Reply:
> I thought the preference is standard type names.

Yes, if they work equally well as the alternative.

`bool` marshalling has always been bug prone area. The 4-byte bool (`UnmanagedType.Bool`) tends to be the least bug-prone option, It is used for interop in this repo in nearly all cases.

Reply:
Ah, it's still somewhat confusing:
* C23's bool is 1 _bit_
* Rust's U1 type is 1 _bit_
* UnmanagedType.U1 is 1 _byte_
* Swift's Bool is 1 _byte_
* Go's bool is 1 _byte_

Reply:
> It does not look great to use a mix of two different bool types in a single signature. I would either use BOOL for everything (change the arg type to out bool so that the marshalling kicks in) or CLR_BOOL for everything.

That makes sense. I can apply `CLR_BOOL`, but will do it across all these APIs. I'd prefer to have all APIs in this domain be the same. That fact that they were not is how we got here in the first place - `bool` vs `BOOL`.